server/src/postgis/nearest.rs

```diff
@@ -0,0 +1,361 @@
+//! This module contains functions for routing between nodes.
+use crate::grpc::server::grpc_server::{
+    DistanceTo, NearestNeighborRequest, NodeType as GrpcNodeType,
+};
+
+use crate::postgis::NodeType;
+use uuid::Uuid;
+
+/// Possible errors with path requests
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum NNError {
+    /// No path was found
+    NoPath,
+
+    /// Invalid start node
+    InvalidStartNode,
+
+    /// Invalid end node
+    InvalidEndNode,
+
+    /// Invalid limit
+    InvalidLimit,
+
+    /// Invalid range
+    InvalidRange,
+
+    /// Unsupported path type
+    Unsupported,
+
+    /// Could not get client
+    Client,
+
+    /// Unknown error
+    Unknown,
+}
+
+impl std::fmt::Display for NNError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            NNError::NoPath => write!(f, "No path was found."),
+            NNError::InvalidStartNode => write!(f, "Invalid start node."),
+            NNError::InvalidEndNode => write!(f, "Invalid end node."),
+            NNError::InvalidLimit => write!(f, "Invalid limit."),
+            NNError::InvalidRange => write!(f, "Invalid range."),
+            NNError::Unsupported => write!(f, "Unsupported path type."),
+            NNError::Client => write!(f, "Could not get client."),
+            NNError::Unknown => write!(f, "Unknown error."),
+        }
+    }
+}
+
+#[derive(Debug)]
+struct NNRequest {
```

Comment:
I see you now need to create this struct just because you had to [redefine](https://github.com/Arrow-air/svc-gis/blob/87a7899e7f1ec23854c4c894c09ebbfc74fd1530/server/src/postgis/mod.rs#L31) the `NodeType`  to add additional attributes.
You can provide these attributes in the tonic build config instead, so you can re-use the already defined `NodeType` from the proto file and, as a result, don't need to define this struct either.

https://docs.rs/tonic-build/latest/tonic_build/struct.Builder.html#method.field_attribute

example implementations from svc-storage:
https://github.com/Arrow-air/svc-storage/blob/c1bbe62bf12b1a54b2615fdfb080c3f0c261df48/includes/build.rs#L156

Reply:
Removed NNRequest and NodeType, using the ones in grpc.rs

https://github.com/Arrow-air/svc-gis/pull/11/commits/b557e02d0da94c5eea8a912d141c424ae93766a8